scout/server/blueprints/variant/controllers.py

```diff
@@ -194,13 +194,24 @@ def variant(
         }
 
     """
-    if not (institute_obj and case_obj):
-        institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
+
     # If the variant is already collected we skip this part
     if not variant_obj:
         # NOTE this will query with variant_id == document_id, not the variant_id.
         variant_obj = store.variant(variant_id)
 
+    if not (institute_obj and case_obj):
```

Comment:
This code is also in the _lookup_variant function above, would be nice to unify and have a single function handling this code.

Reply:
Ok, I agree, I'll refactor this. Should make the other calls involving variant safer as well.